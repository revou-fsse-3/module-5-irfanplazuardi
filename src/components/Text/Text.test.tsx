import { render, screen } from "@testing-library/react";
import Text from ".";
import "@testing-library/jest-dom/extend-expect";

describe("Text unit testing", () => {
  test("Text content works correctly", () => {
    render(<Text>{"testing text"}</Text>);
    const element = screen.getByText("testing text");
    expect(element).toMatchSnapshot();
  });

  test("Text shows the correct class name", () => {
    render(<Text className="Title">{"testing text"}</Text>);
    const element = screen.getByText("testing text");
    expect(element).toHaveClass("Title");
  });
});
