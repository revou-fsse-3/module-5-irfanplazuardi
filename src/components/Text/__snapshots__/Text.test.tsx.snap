// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text unit testing Text content works correctly 1`] = `
<p>
  testing text
</p>
`;
